File: python/src/wdl_model/model/expressions/wdl_null_literal.py
from __future__ import annotations

from .wdl_expression import ComponentType
from .wdl_expression import WdlExpression


class WdlNullLiteral(WdlExpression):
    def __str__(self) -> str:
        return "null"

    def componentType(self) -> ComponentType:
        return ComponentType.LITERAL

File: java/src/main/java/com/myriad/wdl/model/expressions/WdlNullLiteral.java
package com.myriad.wdl.model.expressions;

public final class WdlNullLiteral implements WdlExpression {

  public WdlNullLiteral() {}

  @Override
  public String toString() {
    return getClass().getSimpleName();
  }

  @Override
  public ComponentType componentType() {
    return ComponentType.NULL_LIT;
  }
}
